lib/chef/resource/user.rb

```diff
@@ -30,130 +30,21 @@ class User < Chef::Resource
       default_action :create
       allowed_actions :create, :remove, :modify, :manage, :lock, :unlock
 
-      def initialize(name, run_context = nil)
-        super
-        @username = name
-        @comment = nil
-        @uid = nil
-        @gid = nil
-        @home = nil
-        @shell = nil
-        @password = nil
-        @system = false
-        @manage_home = false
-        @force = false
-        @non_unique = false
-        @iterations = 27855
-        @salt = nil
-      end
-
-      def username(arg = nil)
-        set_or_return(
-          :username,
-          arg,
-          :kind_of => [ String ]
-        )
-      end
-
-      def comment(arg = nil)
-        set_or_return(
-          :comment,
-          arg,
-          :kind_of => [ String ]
-        )
-      end
-
-      def uid(arg = Chef::NOT_PASSED)
-        set_or_return(
-          :uid,
-          arg,
-          :kind_of => [ String, Integer, NilClass ],
-          :coerce => proc { |x| x || nil }
-        )
-      end
-
-      def gid(arg = Chef::NOT_PASSED)
-        set_or_return(
-          :gid,
-          arg,
-          :kind_of => [ String, Integer, NilClass ],
-          :coerce => proc { |x| x || nil }
-        )
-      end
+      property :username, String, name_property: true
+      property :comment, String
+      property :uid, [ String, Integer, NilClass ], default: nil
+      property :gid, [ String, Integer, NilClass ], default: nil
```

Comment:
both these can just be `property :uid, [ String, Integer ]`

nil is always the default and always nillable, unless you set a non-nil default.

Reply:
Fixed